src/mc-schema-broker.c

```diff
@@ -860,60 +858,56 @@ static bool insert_encryptionInformation(const mc_schema_broker_t *sb,
         goto success;
     }
 
-    if (0 != strcmp(cmd_name, "explain") || cmd_target == MC_TO_MONGOCRYPTD) {
-        // All commands except "explain" and "bulkWrite" expect "encryptionInformation"
-        // at top-level. "explain" sent to mongocryptd expects
-        // "encryptionInformation" at top-level.
-        if (!append_encryptionInformation(sb, cmd_name, cmd, status)) {
+    if (0 == strcmp(cmd_name, "explain")
+        && (cmd_target == MC_CMD_SCHEMAS_FOR_CRYPT_SHARED || cmd_target == MC_CMD_SCHEMAS_FOR_SERVER)) {
+        // The "explain" command is a special case when sent to crypt_shared or the server.
+        // crypt_shared and the server expect "encryptionInformation" nested in the "explain" document. Example:
+        // {
+        //    "explain": {
+        //       "find": "to-crypt_shared-or-server"
+        //       "encryptionInformation": {}
+        //    }
+        // }
+        // mongocryptd expects "encryptionInformation" to be a sibling of the "explain" document. Example:
+        // {
+        //    "explain": { "find": "to-mongocryptd" },
+        //    "encryptionInformation": {}
+        // }
```

Comment:
```suggestion
        // The "explain" command is a special case when sent to crypt_shared or the server, which expect
        // "encryptionInformation" nested in the "explain" document instead of at top-level. Example:
        // {
        //    "explain": {
        //       "find": "to-crypt_shared-or-server"
        //       "encryptionInformation": {}
        //    }
        // }
```
Suggest removing mention of mongocryptd for "explain", which is no different to the default case handled below (special case description only need apply to the special case).